src/main/java/uk/gov/hmcts/divorce/citizen/event/CitizenFinalOrderDelayReason.java

```diff
@@ -51,14 +52,22 @@ public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(CaseDetails<C
         log.info("Citizen final order delay reason about to submit callback invoked for Case Id: {}", details.getId());
 
         CaseData data = details.getData();
+        State endState = FinalOrderRequested;
 
         if (data.getApplicationType().isSole()) {
             notificationDispatcher.send(soleAppliedForFinalOrderNotification, data, details.getId());
         }
 
+        if (data.isWelshApplication()) {
+            data.getApplication().setWelshPreviousState(endState);
+            endState = WelshTranslationReview;
+            log.info("State set to WelshTranslationReview, WelshPreviousState set to {}, CaseID {}",
+                data.getApplication().getWelshPreviousState(), details.getId());
+        }
+
         return AboutToStartOrSubmitResponse.<CaseData, State>builder()
             .data(data)
-            .state(FinalOrderRequested)
+            .state(endState)
```

Comment:
this can then be set as `WelshTranslationReview`

Reply:
changing the state here would set it as WelshTranslationReview for all cases on this event regardless of the WelshLanguage fields. Although we could split this up into using two separate returns depending on which state to set it to

Reply:
thanks for clarifying that makes sense